packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1942,7 +1904,7 @@ export class ImportResolver {
                             }
                         } else if (innerEntry.isFile()) {
                             if (innerEntry.name.endsWith('.pyi')) {
-                                const strippedFileName = stripFileExtension(innerEntry.name);
+                                const strippedFileName = innerEntry.name.slice(0, -3);
```

Comment:
Shouldn't this be -4 instead of -3?

Also, shouldn't this be removing the extension in the same way that `_buildStdlibCache` does for `.py`. I'm not in love with either approach, but they should be the same?

Reply:
Yeah this looks like a bug. I should put it back to the way it was before.